authentication/models/user.py

```diff
@@ -135,8 +135,15 @@ def years_since_entry(self):
 
             return today.year - self.year_of_entry - 1
 
-    def get_role(self, association):
-        for role in self.roles.all():
-            if role.association == association:
-                return role
+    def get_role(self, association=None, association_pk=None):
+        q = None
+
+        if association_pk is not None:
+            q = self.roles.filter(association_id=association_pk)
+        elif association is not None:
+            q = self.roles.filter(association=association)
```

Comment:
I beleive you can use `self.roles.filter(association=association)` if `association` is either an `Association` or the association id

Reply:
Indeed!